packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/entity_type/read.rs

```diff
@@ -0,0 +1,142 @@
+            [segment, segments @ ..] => {
+                // TODO: Avoid cloning on literals
+                //   see https://app.asana.com/0/0/1202884883200947/f
+                let literal = match segment.identifier.as_str() {
+                    "uri" => Literal::String(self.record.id().base_uri().to_string()),
+                    "version" => Literal::Version(self.record.id().version(), self.is_latest),
+                    "title" => Literal::String(self.record.title().to_owned()),
+                    "description" => self
+                        .record
+                        .description()
+                        .map_or(Literal::Null, |description| {
+                            Literal::String(description.to_owned())
+                        }),
+                    "default" | "examples" => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                    "properties" => match segments {
+                        [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                        [property_type_segment, property_type_segments @ ..]
+                            if property_type_segment.identifier == "**" =>
+                        {
+                            // TODO: Use relation tables
+                            //   see https://app.asana.com/0/0/1202884883200942/f
+                            return Ok(Literal::List(
+                                stream::iter(self.record.property_type_references())
+                                    .then(|property_type_ref| async {
+                                        context
+                                            .read_versioned_property_type(property_type_ref.uri())
+                                            .await
+                                            .change_context(ResolveError::StoreReadError)?
+                                            .resolve(property_type_segments, context)
+                                            .await
+                                    })
+                                    .try_collect()
+                                    .await?,
+                            ));
+                        }
+                        [property_type_segment, ..] if property_type_segment.identifier == "*" => {
+                            todo!("{}", UNIMPLEMENTED_WILDCARDS)
+                        }
+                        _ => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                    },
+                    "required" => Literal::List(
+                        self.record
+                            .required()
+                            .iter()
+                            .map(|base_uri| Literal::String(base_uri.to_string()))
+                            .collect(),
+                    ),
+                    "links" => match segments {
+                        [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                        [entity_type_segment, entity_type_segments @ ..] => {
```

Comment:
Would these be `link_type_segment*` conceptually? (i.e. should we rename these match patterns to reflect that)

Reply:
I solved this by moving out the resolving in bd7a71362ea09badf16355fd7824b3d6cad134db
I will apply this to the other implementations in my follow-up.